### DISCUSSION TITLE
link file to book

### DISCUSSION THREAD

**Discussion by coder-ant** [TortugaPower/BookPlayer#1014] [Q&A]
Hi, I had earlier deleted a .m4b file in the iOS app to free up space on the device. I did not delete the corresponding book itself hoping to preserve the bookmarks.
Now when I add the file back, it creates a new book instead of linking to existing book. Is there a way to link the added file to an existing book so that the bookmarks are preserved

**Comment by GianniCarlo** [comment]
Hi @coder-ant ! Yes this should have worked out of the box, we use the file name to search and try to link if there's already a dangling book. What I think could have happened is that for the original book, at the moment it was imported, the suffix '-1'  was added before the file extension either by iOS (could have happened via airdrop) or by the app, when copying the item to the Processed folder we try to use the original file name and if it's already taken by another file in there we increment the suffix.

That said, I'll include in the next beta the registered file name we have for each book inside their details page, I'm concluding some tests about adopting app intents in preparation for interactive widgets, but I'll ping back when the beta is updated with this changes, as you'd be able to verify the exact name, and just import the file with that name and auto linking should work

**Reply by coder-ant** [reply]
cool, thanks.

**Reply by GianniCarlo** [reply]
@coder-ant the new version 5.0.10 is out now on the AppStore, I forgot to ping back when I put the build on the beta, but hopefully you saw the alert. In any case, let me know if it helped having the original filename at the bottom of the item details screen

**Reply by coder-ant** [reply]
Yeah, just saw it. That was quick and looks good. Thanks.
